File: mobile/lib/features/schedule/domain/enum/schedule_recurrence_type.dart
enum ScheduleRecurrenceType { none, daily, weekly, customWeekly, monthly }

extension ScheduleRecurrenceTypeX on ScheduleRecurrenceType {
  static ScheduleRecurrenceType fromApi(String value) {
    switch (value.toUpperCase()) {
      case 'DAILY':
        return ScheduleRecurrenceType.daily;
      case 'WEEKLY':
        return ScheduleRecurrenceType.weekly;
      case 'CUSTOM_WEEKLY':
        return ScheduleRecurrenceType.customWeekly;
      case 'MONTHLY':
        return ScheduleRecurrenceType.monthly;
      default:
        return ScheduleRecurrenceType.none;
    }
  }

  String get apiValue {
    switch (this) {
      case ScheduleRecurrenceType.none:
        return 'NONE';
      case ScheduleRecurrenceType.daily:
        return 'DAILY';
      case ScheduleRecurrenceType.weekly:
        return 'WEEKLY';
      case ScheduleRecurrenceType.customWeekly:
        return 'CUSTOM_WEEKLY';
      case ScheduleRecurrenceType.monthly:
        return 'MONTHLY';
    }
  }

  String get label {
    switch (this) {
      case ScheduleRecurrenceType.none:
        return 'Once';
      case ScheduleRecurrenceType.daily:
        return 'Daily';
      case ScheduleRecurrenceType.weekly:
        return 'Weekly';
      case ScheduleRecurrenceType.customWeekly:
        return 'Custom weekly';
      case ScheduleRecurrenceType.monthly:
        return 'Monthly';
    }
  }
}

File: backend/src/main/java/com/lifeos/backend/schedule/domain/enums/ScheduleRecurrenceType.java
package com.lifeos.backend.schedule.domain.enums;

/**
 * Defines how a ScheduleTemplate repeats.
 *
 * Schedule recurrence belongs to ScheduleTemplate.
 * ScheduleOccurrence is the real spawned time block.
 */
public enum ScheduleRecurrenceType {

    /**
     * Happens once on recurrenceStartDate.
     */
    NONE,

    /**
     * Repeats every day.
     */
    DAILY,

    /**
     * Repeats weekly on the same day as recurrenceStartDate.
     */
    WEEKLY,

    /**
     * Repeats on selected weekdays.
     *
     * Example:
     * MONDAY,WEDNESDAY,FRIDAY
     */
    CUSTOM_WEEKLY,

    /**
     * Repeats monthly on the same day-of-month as recurrenceStartDate.
     */
    MONTHLY;

    public boolean isRecurring() {
        return this != NONE;
    }

    public boolean isOneTime() {
        return this == NONE;
    }

    public boolean requiresStartDate() {
        return true;
    }

    public boolean requiresDaysOfWeek() {
        return this == CUSTOM_WEEKLY;
    }

    public boolean usesStartDateDayOfWeek() {
        return this == WEEKLY;
    }

    public boolean usesStartDateDayOfMonth() {
        return this == MONTHLY;
    }
}